docs/content/docs/build/auth.md

```diff
@@ -101,13 +101,50 @@ provider is useful for testing, but in production scenarios you must use a secur
 
 ### A secure serverless token provider
 
-One option for building a secure token provider is to create a serverless Azure Function and expose it as a token
-provider. This enables you to store the *tenant secret key* on a secure server. Your application calls the Azure Function to
-generate tokens rather than signing them locally like the `InsecureTokenProvider` does.
+One option for building a secure token provider is to create a serverless Azure Function and expose it. This enables you to store the *tenant secret key* on a secure server. Your application calls the Azure Function to generate tokens rather than signing them locally like the `InsecureTokenProvider` does.
+
+You can create the token provider class say `AzureFunctionTokenProvider` which extends the `ITokenProvider` interface without exposing the tenant key secret in client-side code. This class would be responsible for fetching the token from your very own backend. It accepts the url to Azure function and an optional user object.
+
+```typescript
+import { ITokenProvider, ITokenResponse } from "@fluidframework/azure-client";
+
+export class AzureFunctionTokenProvider implements ITokenProvider {
+  constructor(
+    private readonly azFunctionUrl: string,
+    private readonly user?: Pick<AzureMember, "userId" | "userName" | "additionalDetails">,
+  );
+
+  fetchOrdererToken(tenantId: string, documentId: string, refresh?: boolean): Promise<ITokenResponse>;
+
+  fetchStorageToken(tenantId: string, documentId: string, refresh?: boolean): Promise<ITokenResponse>;
+}
+```
+
+`fetchOrdererToken` and `fetchStorageToken` are responsible for fetching the orderer and storgae url from the host respectively. They return `TokenResponse` object representing token value along with flag indicating whether token came from cache.
+
+To ensure that the secret doesn't get exposed, it is passed to a secure, backend Azure function from which the token is fetched. One of the possible way to fetch the token is making an axios `GET` request call to your Azure function by passing in the tenantID, documentId and userID/userName as optional parameters. Azure function is responsible for mapping between the tenant ID to a tenant key secret to generate and sign the token such that the service will accept it.
+
+```typescript
+private async getToken(tenantId: string, documentId: string): Promise<string> {
+  return axios.get(this.azFunctionUrl, {
```

Comment:
I don't have any problem with axios, but is there a vanilla JS way to do this that isn't ugly or illegible? If not, then I suggest adding a callout above this code sample that says, "The example below uses the axios (add a link to the npm package) library to make HTTP requests. You can use other libraries or approaches to making an HTTP request."

Reply:
^^ This is why my suggestion was to have this as a `Recipe` or something talking about how to build a specific scenario. Then we have a more basic doc that talks about the principals where you can sudo code something like:

```js
private async getToken(tenantId, documentId) {
  // make call to backend service to retrieve server generated token 
  const token = getTokenFromServer(...);
  return token;
```